src/GraphQL/Types/ArgumentInformation.cs

```diff
@@ -18,29 +19,34 @@ public class ArgumentInformation
         /// <summary>
         /// Initializes a new instance with the specified parameters.
         /// </summary>
-        public ArgumentInformation(ParameterInfo parameterInfo, Type sourceType, FieldType fieldType, TypeInformation typeInformation, LambdaExpression? expression)
+        public ArgumentInformation(ParameterInfo parameterInfo, Type? sourceType, FieldType? fieldType, TypeInformation typeInformation, LambdaExpression? expression)
         {
             ParameterInfo = parameterInfo ?? throw new ArgumentNullException(nameof(parameterInfo));
-            FieldType = fieldType ?? throw new ArgumentNullException(nameof(fieldType));
-            SourceType = sourceType ?? throw new ArgumentNullException(nameof(sourceType));
+            FieldType = fieldType; // ?? throw new ArgumentNullException(nameof(fieldType));
+            SourceType = sourceType; // ?? throw new ArgumentNullException(nameof(sourceType));
             TypeInformation = typeInformation ?? throw new ArgumentNullException(nameof(typeInformation));
             Expression = expression;
         }
 
         /// <summary>
         /// Initializes a new instance with the specified parameters.
-        /// If the parameter type is <see cref="IResolveFieldContext"/> or <see cref="CancellationToken"/>,
+        /// If the parameter type is <see cref="IResolveFieldContext"/>, <see cref="IResolveEventStreamContext"/> or <see cref="CancellationToken"/>,
         /// an expression is generated for the parameter and set within <see cref="Expression"/>; otherwise
         /// <see cref="Expression"/> is set to <see langword="null"/>.
         /// </summary>
-        public ArgumentInformation(ParameterInfo parameterInfo, Type sourceType, FieldType fieldType, TypeInformation typeInformation)
+        public ArgumentInformation(ParameterInfo parameterInfo, Type? sourceType, FieldType? fieldType, TypeInformation typeInformation)
             : this(parameterInfo, sourceType, fieldType, typeInformation, null)
         {
             if (parameterInfo.ParameterType == typeof(IResolveFieldContext))
             {
                 Expression<Func<IResolveFieldContext, IResolveFieldContext>> expr = x => x;
                 Expression = expr;
             }
+            else if (parameterInfo.ParameterType == typeof(IResolveEventStreamContext))
+            {
+                Expression<Func<IResolveFieldContext, IResolveEventStreamContext>> expr = x => (IResolveEventStreamContext)x;
+                Expression = expr;
+            }
```

Comment:
Not ideal, but to resolve an `IResolveEventStreamContext` as a method argument the context gets implicitly cast to `IResolveFieldContext` and then explicitly cast back to `IResolveEventStreamContext`.